modular_skyrat/modules/modular_items/lewd_items/code/arousal_system.dm

```diff
@@ -0,0 +1,968 @@
+	climax(TRUE)
+
+/mob/living/proc/set_masochism(status) //TRUE or FALSE
+	if(status == TRUE)
+		masochism = status
+		pain_limit = 80
+	if(status == FALSE)
+		masochism = status
+		pain_limit = 20
+
+/mob/living/proc/set_nymphomania(status) //TRUE or FALSE
+	if(status == TRUE)
+		nymphomania = TRUE
+	if(status == FALSE)
+		nymphomania = FALSE
+
+/mob/living/proc/set_neverboner(status) //TRUE or FALSE
+	if(status == TRUE)
+		neverboner = TRUE
+	if(status == FALSE)
+		neverboner = FALSE
+
+////////////
+///FLUIDS///
+////////////
+
+/datum/status_effect/body_fluid_regen
+	id = "body fluid regen"
+	tick_interval = 50
+	duration = -1
+	alert_type = null
+
+/datum/status_effect/body_fluid_regen/tick()
+	var/mob/living/carbon/human/H = owner
+	if(owner.stat != DEAD && H.client?.prefs.erp_pref == "Yes")
+		var/obj/item/organ/genital/testicles/balls = owner.getorganslot(ORGAN_SLOT_TESTICLES)
+		var/obj/item/organ/genital/breasts/breasts = owner.getorganslot(ORGAN_SLOT_BREASTS)
+		var/obj/item/organ/genital/vagina/vagina = owner.getorganslot(ORGAN_SLOT_VAGINA)
+
+		var/interval = 5
+		if(balls)
+			if(owner.arousal >= AROUS_SYS_LITTLE)
+				var/regen = (owner.arousal/100) * (balls.internal_fluids.maximum_volume/235) * interval
+				balls.internal_fluids.add_reagent(/datum/reagent/consumable/cum, regen)
+
+		if(breasts)
+			if(breasts.lactates == TRUE)
+				var/regen = ((owner.nutrition / (NUTRITION_LEVEL_WELL_FED/100))/100) * (breasts.internal_fluids.maximum_volume/11000) * interval
+				breasts.internal_fluids.add_reagent(/datum/reagent/consumable/milk/breast_milk, regen)
+				if(!breasts.internal_fluids.holder_full())
+					owner.adjust_nutrition(regen / 2)
+				else
+					regen = regen
+
+		if(vagina)
+			if(owner.arousal >= AROUS_SYS_LITTLE)
+				var/regen = (owner.arousal/100) * (vagina.internal_fluids.maximum_volume/250) * interval
+				vagina.internal_fluids.add_reagent(/datum/reagent/consumable/girlcum, regen)
+				if(vagina.internal_fluids.holder_full() && regen >= 0.15)
+					regen = regen
+			else
+				vagina.internal_fluids.remove_any(0.05)
+
+/////////////
+///AROUSAL///
+/////////////
+/mob/living/proc/get_arousal()
+	return arousal
+
+/mob/living/proc/adjustArousal(arous = 0)
+	if(stat != DEAD && client?.prefs.erp_pref == "Yes")
+		arousal += arous
+
+		var/arousal_flag = AROUSAL_NONE
+		if(arousal >= AROUS_SYS_STRONG)
+			arousal_flag = AROUSAL_FULL
+		else if(arousal >= AROUS_SYS_LITTLE)
+			arousal_flag = AROUSAL_PARTIAL
+
+		if(arousal_status != arousal_flag) // Set organ arousal status
+			arousal_status = arousal_flag
+			if(istype(src,/mob/living/carbon/human))
+				var/mob/living/carbon/human/M = src
+				for(var/i=1,i<=M.internal_organs.len,i++)
+					if(istype(M.internal_organs[i],/obj/item/organ/genital))
+						var/obj/item/organ/genital/G = M.internal_organs[i]
+						if(!G.aroused == AROUSAL_CANT)
+							G.aroused = arousal_status
+							G.update_sprite_suffix()
+				M.update_body()
+	else
+		arousal -= abs(arous)
```

Comment:
This is just to implement significant names to your variables, Gandalf's comment still stands.
```suggestion
/mob/living/proc/adjustArousal(new_arousal = 0)
	if(stat != DEAD && client?.prefs.erp_pref == "Yes")
		arousal += new_arousal

		var/arousal_flag = AROUSAL_NONE
		if(arousal >= AROUS_SYS_STRONG)
			arousal_flag = AROUSAL_FULL
		else if(arousal >= AROUS_SYS_LITTLE)
			arousal_flag = AROUSAL_PARTIAL

		if(arousal_status != arousal_flag) // Set organ arousal status
			arousal_status = arousal_flag
			if(istype(src,/mob/living/carbon/human))
				var/mob/living/carbon/human/target = src
				for(var/i=1,i<=target.internal_organs.len,i++)
					if(istype(target.internal_organs[i],/obj/item/organ/genital))
						var/obj/item/organ/genital/genital = target.internal_organs[i]
						if(genital.aroused != AROUSAL_CANT)
							genital.aroused = arousal_status
							genital.update_sprite_suffix()
				target.update_body()
	else
		arousal -= abs(new_arousal) 
```